Add aws-noaa-oisst-avhrr-only example notebook.

sharkinsspatial opened the pull request.

sharkinsspatial committed:
Add initial example for aws-noaa-oisst-avhrr-only.

sharkinsspatial committed:
Add narrative to aws-noaa-oisst-avhrr example and pip install libs.

wildintellect commented on examples/aws-noaa-oisst-avhrr-only-example.ipynb:
```diff
@@ -0,0 +1,936 @@
+{
```
<p>Needs a top level title and intro that is about the Dataset before explaining that this dataset is accessible thanks to kerchunk and pangeo.</p><p> </p><p>In some of the later sections the Cloud Optimized information comes first, but should probably come after the description of what the user is actually trying to do. Overall the orientation should be more focused on SST data with the technical parts there to explain how Cloud Optimized access works. While this is the first example of the Kerchunk method we expect to do many.</p><p> </p><p>Also added suggested Comment lines for some of the code blocks, to make sure people know what each one does.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/asdi-examples/pull/8/discussion/#comment-1170378136'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

wildintellect replied:
<p>can we write out words instead of <code>repr</code> I'm not sure that's going to be clear to people?</p><p> </p><blockquote>the full 40 years of data.</blockquote><p>", from 1981 to 2022."</p><p>Make it clear the printout doesn't actually say 40 anywhere, but that derived from reading the values.</p><p> </p><p><code>remote_protocol</code> isn't mentioned in the text</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/asdi-examples/pull/8/discussion/#comment-1170378143'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='6'/>

wildintellect replied:
<pre class="ql-syntax" spellcheck="false"># Select Sea Surface Temperature at a single point over 1 month
Line #1.    point = ds["sst"].sel(lat=10, lon=100, method="nearest")
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/asdi-examples/pull/8/discussion/#comment-1170378144'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='9'/>

wildintellect replied:
<pre class="ql-syntax" spellcheck="false"># Select Sea Surface temperature for a single data and region
# Region defined by setting mim, max coordinates with slice
Line #1.    subset = ds["sst"].sel(time="2019-05-15",lat=slice(-10,30),lon=slice(70,110))

</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/asdi-examples/pull/8/discussion/#comment-1170378148'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

wildintellect commented on examples/aws-noaa-oisst-avhrr-only-example.ipynb:
```diff
@@ -0,0 +1,936 @@
+{
```
<p>Should we pin pystac_client to newer than a certain value? <code>&gt;0.6.0</code>?make sure to quote to escape shell interpretation.</p><p>Change <code>!</code> to <code>%</code> per jupyter/SMSL recommendations</p><p> </p><p>I'm looking to why I got a pip error</p><pre class="ql-syntax" spellcheck="false">ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
boto3 1.26.110 requires botocore&lt;1.30.0,&gt;=1.29.110, but you have botocore 1.29.76 which is incompatible.
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/developmentseed/asdi-examples/pull/8/discussion/#comment-1170378141'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

wildintellect replied:
Works fine in a fresh conda env. TODO for later is general addition of using Conda envs.

wildintellect commented:
Also missing, add a line to the Readme that links to the SMSL launcher for where this notebook will be once the PR is merged.

wildintellect committed:
chore: relocate/rename noaa oisst notebook

wildintellect committed:
feat: Readability enhancements to #8

wildintellect committed:
Merge branch 'main' into aws-noaa-oisst-avhrr-only_example

wildintellect committed:
feat: Added noaa-oisst to list of notebooks

Also updated notebook descriptions to include the type of Cloud Optimized access

wildintellect committed:
feat: prototype conda yaml

This is a generic Zarr so might become the same for many tutorials, and might be moved to a common environments folder

wildintellect committed:
chore: Conda env name to match yaml

jjfrench commented on noaa-oisst/noaa-oisst-avhrr-only.ipynb:
```diff
@@ -0,0 +1,714 @@
+   "id": "679992a4-0179-4d11-95f0-112d2f91ab83",
+   "metadata": {},
+   "source": [
+    "### Xarray selection operations\n",
```
```suggestion
    "### Xarray Selection Operations\n",
```

jjfrench commented on noaa-oisst/noaa-oisst-avhrr-only.ipynb:
```diff
@@ -0,0 +1,714 @@
+   "id": "e738dd81-4d2c-4a44-8c7a-357776882d69",
+   "metadata": {},
+   "source": [
+    "### Opening the kerchunk reference file\n",
```
```suggestion
    "### Opening the Kerchunk Reference File\n",
```

jjfrench commented on noaa-oisst/noaa-oisst-avhrr-only.ipynb:
```diff
@@ -0,0 +1,714 @@
+    "tags": []
+   },
+   "source": [
+    "### Using the notebook\n",
```
```suggestion
    "### Using the Notebook\n",
```

jjfrench commented on noaa-oisst/noaa-oisst-avhrr-only.ipynb:
```diff
@@ -0,0 +1,714 @@
+    "### How the Reference File was made\n",
+    "To generate a reference file for the [NOAA OISST](https://registry.opendata.aws/noaa-cdr-oceanic/) data files we leverage the community driven [Pangeo Forge](https://pangeo-forge.readthedocs.io/en/latest/what_is_pangeo_forge.html) project.  This involves creating a [Pangeo Forge recipe](https://github.com/pangeo-forge/staged-recipes/pull/232) which is [run](https://pangeo-forge.org/dashboard ...
+    "\n",
+    "> The [Kerchunk](https://fsspec.github.io/kerchunk/) library allows exposing the internal chunk layout of large numbers of gridded scientifc data files as a single virtual [Zarr](https://zarr.readthedocs.io/en/stable/index.html) dataset.  This virtual dataset allows users to interact with the data in cloud optimized way by accessing a single Kerchunk [reference file](https://fsspec.github.io ...
```
```suggestion
    "> The [Kerchunk](https://fsspec.github.io/kerchunk/) library allows exposing the internal chunk layout of large numbers of gridded scientific data files as a single virtual [Zarr](https://zarr.readthedocs.io/en/stable/index.html) dataset.  This virtual dataset allows users to interact with the data in cloud optimized way by accessing a single Kerchunk [reference file](https://fsspec.github.io/kerchunk/spec.html) entry point without altering or duplicating the underlying source data files.\n",
```

jjfrench commented on noaa-oisst/noaa-oisst-avhrr-only.ipynb:
```diff
@@ -0,0 +1,714 @@
+    "[NOAA Sea Surface Temperature - Optimum Interpolation](https://registry.opendata.aws/noaa-cdr-oceanic/) also known as NOAA OISST, is distributed as a time series of NetCDF formatted HDF5 files. In order to use those files in a cloud optimized way, where we can subset the data by location or time, we can wrap in them with an index (Zarr reference file). Using the AWS STAC API we can discover  ...
+    "\n",
+    "### How the Reference File was made\n",
+    "To generate a reference file for the [NOAA OISST](https://registry.opendata.aws/noaa-cdr-oceanic/) data files we leverage the community driven [Pangeo Forge](https://pangeo-forge.readthedocs.io/en/latest/what_is_pangeo_forge.html) project.  This involves creating a [Pangeo Forge recipe](https://github.com/pangeo-forge/staged-recipes/pull/232) which is [run](https://pangeo-forge.org/dashboard ...
```
```suggestion
    "To generate a reference file for the [NOAA OISST](https://registry.opendata.aws/noaa-cdr-oceanic/) data files, we leverage the community-driven [Pangeo Forge](https://pangeo-forge.readthedocs.io/en/latest/what_is_pangeo_forge.html) project.  This involves creating a [Pangeo Forge recipe](https://github.com/pangeo-forge/staged-recipes/pull/232) which is [run](https://pangeo-forge.org/dashboard/feedstock/93) using the Pangeo Forge bakery infrastructure that creates an open reference file which points to the underlying data [files](https://noaa-cdr-sea-surface-temp-optimum-interpolation-pds.s3.amazonaws.com/index.html) on S3 made available  via the [Amazon Sustainability Data Initiative](https://sustainability.aboutamazon.com/environment/the-cloud/amazon-sustainability-data-initiative).\n",
```

jjfrench commented on noaa-oisst/noaa-oisst-avhrr-only.ipynb:
```diff
@@ -0,0 +1,714 @@
+    "\n",
+    "[NOAA Sea Surface Temperature - Optimum Interpolation](https://registry.opendata.aws/noaa-cdr-oceanic/) also known as NOAA OISST, is distributed as a time series of NetCDF formatted HDF5 files. In order to use those files in a cloud optimized way, where we can subset the data by location or time, we can wrap in them with an index (Zarr reference file). Using the AWS STAC API we can discover  ...
+    "\n",
+    "### How the Reference File was made\n",
```
```suggestion
    "### How the Reference File Was Made\n",
```

jjfrench commented on noaa-oisst/noaa-oisst-avhrr-only.ipynb:
```diff
@@ -0,0 +1,714 @@
+    "\n",
+    "## Accessing as a Cloud Native Virtual Dataset\n",
+    "\n",
+    "[NOAA Sea Surface Temperature - Optimum Interpolation](https://registry.opendata.aws/noaa-cdr-oceanic/) also known as NOAA OISST, is distributed as a time series of NetCDF formatted HDF5 files. In order to use those files in a cloud optimized way, where we can subset the data by location or time, we can wrap in them with an index (Zarr reference file). Using the AWS STAC API we can discover  ...
```
```suggestion
    "[NOAA Sea Surface Temperature - Optimum Interpolation](https://registry.opendata.aws/noaa-cdr-oceanic/) also known as NOAA OISST, is distributed as a time series of NetCDF formatted HDF5 files. To use those files in a cloud-optimized way, where we can subset the data by location or time, we wrap them with an index (Zarr reference file). Using the AWS STAC API we can discover this virtual Zarr dataset and interact with the underlying data files in a cloud-optimized manner.\n",
```

wildintellect committed:
Title Case

Co-authored-by: Jamison French <50224594+jjfrench@users.noreply.github.com>

wildintellect committed:
Title Case

Co-authored-by: Jamison French <50224594+jjfrench@users.noreply.github.com>

wildintellect committed:
Title Case

Co-authored-by: Jamison French <50224594+jjfrench@users.noreply.github.com>

wildintellect committed:
Title Case

Co-authored-by: Jamison French <50224594+jjfrench@users.noreply.github.com>

wildintellect committed:
Typos

Co-authored-by: Jamison French <50224594+jjfrench@users.noreply.github.com>

wildintellect committed:
minor: typo

Co-authored-by: Jamison French <50224594+jjfrench@users.noreply.github.com>

wildintellect committed:
minor: word smithing

Co-authored-by: Jamison French <50224594+jjfrench@users.noreply.github.com>

jjfrench reviewed (approved):
LGTM! 👍🏼
